AsyncDisplayKit/TextKit/ASTextKitTailTruncater.mm

```diff
@@ -152,9 +152,14 @@ - (NSUInteger)_findTruncationInsertionPointAtOrBeforeCharacterIndex:(NSUInteger)
 
 - (void)truncate
 {
+  // Reset the text storage to start always with the full string
+  [_context resetTextStorage];
```

Comment:
It seems to me that because the code below is responsible for modifying the string, this object should probably also be responsible for ensuring that it is working from the original string before truncating.  Can we have this object obtain the string it truncates from a delegate or something, probably the ASTextNode itself, so the data flow is always in the same direction / at the same level?

Maybe this is a bad idea.  This is largely an existing architectural problem, not just a problem with the solution - we're essentially talking about how to adjust the solution to be as clear as possible considering the architecture is a bit strange here.  Open to any suggestions, happy to land this without changes / just adding comments if you are confident it is the best structure.